In [2]:
import pandas as pd
import numpy as np

In [4]:
file_path = "../static/data/scaled_indicators.csv"
global_df = pd.read_csv(file_path, sep=";")

In [5]:
global_df.head()

Unnamed: 0  Unnamed: 0.1         name     indicator  \
0           0             0  Afghanistan  Aid received   
1           1             1  Afghanistan  Aid received   
2           2             2  Afghanistan  Aid received   
3           3             3  Afghanistan  Aid received   
4           4             4  Afghanistan  Aid received   

                                indicator2  year        value region  \
0  ODA received, total (constant 2010 US$)  1960  108800000.0   Asia   
1  ODA received, total (constant 2010 US$)  1961  223030000.0   Asia   
2  ODA received, total (constant 2010 US$)  1962  110970000.0   Asia   
3  ODA received, total (constant 2010 US$)  1963  225130000.0   Asia   
4  ODA received, total (constant 2010 US$)  1964  290570000.0   Asia   

  predicted  value_scaled  
0     False      3.445057  
1     False      7.169534  
2     False      3.780866  
3     False      3.744915  
4     False      3.728892

In [197]:
global_df[(global_df["name"] == "Afghanistan") & (global_df["indicator"] == "Aid received")
                                  & (global_df["year"] == int(1960))][["value", "year"]].values[0].tolist()

[108800000.0, 1960.0]

## Description des indicateurs

In [22]:
list_indicators = global_df["indicator"].unique()

for indicator in list_indicators:
    min_value = global_df.loc[global_df["indicator"] == indicator, ["value"]].min().values[0]
    max_value = global_df.loc[global_df["indicator"] == indicator, ["value"]].max().values[0]
    mean_value = global_df.loc[global_df["indicator"] == indicator, ["value"]].mean().values[0]

    print("Range indicateur {} : [{}, {}], moyenne : {}".format(indicator, min_value, max_value, mean_value))

Range indicateur Aid received : [-1143420000.0, 24858710000.0], moyenne : 425425935.21594685
Range indicateur Exports (% of GDP) : [0.182968803942039, 241.402014528153], moyenne : 36.15009428953285
Range indicateur External debt : [0.0, 1210.05989933245], moyenne : 52.9829149375682
Range indicateur Total Gross Domestic Product (GDP), Purchasing Power Parity (PPP) : [1051405.91398389, 15664618589539.4], moyenne : 52114833320.13736
Range indicateur Aid received % of GNI (Gross National Income) : [-0.745017151425095, 242.286413767446], moyenne : 8.186217408032645
Range indicateur Total number of billionaires : [0.0, 57.0], moyenne : 1.5766331658291457
Range indicateur Poverty headcount ratio at $2 a day (PPP) (% of population) : [0.0, 98.45], moyenne : 42.66121356421357
Range indicateur Inflation : [-49.4222790062954, 15442.3007178263], moyenne : 29.624382172433165
Range indicateur Investment : [-17.3755100679111, 113.577855400879], moyenne : 23.10481313455939
Range indicateur Tax revenue

In [23]:
list_indicators = global_df_scaled["indicator"].unique()

for indicator in list_indicators:
    min_value = global_df_scaled.loc[global_df_scaled["indicator"] == indicator, ["value"]].min().values[0]
    max_value = global_df_scaled.loc[global_df_scaled["indicator"] == indicator, ["value"]].max().values[0]
    mean_value = global_df_scaled.loc[global_df_scaled["indicator"] == indicator, ["value"]].mean().values[0]

    print("Range indicateur {} : [{}, {}], moyenne : {}".format(indicator, min_value, max_value, mean_value))

Range indicateur Aid received : [-1143420000.0, 24858710000.0], moyenne : 425425935.21594685
Range indicateur Exports (% of GDP) : [0.182968803942039, 241.402014528153], moyenne : 36.15009428953285
Range indicateur External debt : [0.0, 1210.05989933245], moyenne : 52.9829149375682
Range indicateur Total Gross Domestic Product (GDP), Purchasing Power Parity (PPP) : [1051405.91398389, 15664618589539.4], moyenne : 52114833320.13736
Range indicateur Aid received % of GNI (Gross National Income) : [-0.745017151425095, 242.286413767446], moyenne : 8.186217408032645
Range indicateur Total number of billionaires : [0.0, 57.0], moyenne : 1.5766331658291457
Range indicateur Poverty headcount ratio at $2 a day (PPP) (% of population) : [0.0, 98.45], moyenne : 42.66121356421357
Range indicateur Inflation : [-49.4222790062954, 15442.3007178263], moyenne : 29.624382172433165
Range indicateur Investment : [-17.3755100679111, 113.577855400879], moyenne : 23.10481313455939
Range indicateur Tax revenue

In [12]:
indicator_to_rescale = ['Aid received',
'Exports (% of GDP)',
'External debt',
'Total Gross Domestic Product (GDP), Purchasing Power Parity (PPP)',
'Aid received % of GNI (Gross National Income)',
'Total number of billionaires',
'Inflation',
'Investment',
'Tax revenue (% of GDP)',
'Boys primary completion rate (% of boys)',
'Girls primary school completion rate (% of girls)',
'Primary completion rate, total (% of relevant age group)',
'Expenditure per student, primary (% of GDP per capita)',
'Expenditure per student, secondary (% of GDP per capita)',
'Expenditure per student, tertiary (% of GDP per capita)',
'Ratio of girls to boys in primary and secondary education (%)',
'Years in school men 25+',
'Years in school women 25+',
'Official Development Assistance (ODA) % Gross National Income (GNI )']

"Aid received" => si NaN c'est qu'il n'y a probablement pas d'aide, donc ne pas compléter
    
"Exports (p of GDP)" => Pourcentage de 0.18 a 241, à mettre à l'échelle

"External debt" => de 0 à 1210, à mettre à l'échelle

"indicator gdp_total_ppp" => de 10e6 à 10e13, à mettre à l'échelle

...

## Prédiction des valeurs manquantes

In [69]:
file_path = "../static/data/gini_full.csv"
global_df = pd.read_csv(file_path)

In [70]:
global_df["predicted"] = False
global_df.loc[global_df["value"].isnull(), ["predicted"]] = True

In [64]:
global_df["predicted"].value_counts()

True     17645
False     2985
Name: predicted, dtype: int64

In [65]:
global_df.head()

Unnamed: 0            name indicator  indicator2  year value predicted
0           0     Afghanistan      gini  GINI index  1978   NaN      True
1           1         Albania      gini  GINI index  1978   NaN      True
2           2         Algeria      gini  GINI index  1978   NaN      True
3           3  American Samoa      gini  GINI index  1978   NaN      True
4           4         Andorra      gini  GINI index  1978   NaN      True

In [74]:
len(global_df)

14451

In [73]:
global_df = global_df.drop_duplicates(subset=["name", "year"])

In [78]:
global_df.loc[global_df["value"]==" Dem. Rep.", ["value"]] = np.NaN

In [82]:
global_df.loc[global_df["value"]==" Rep.", ["value"]] = np.NaN

In [66]:
list_indicators = global_df["indicator"].unique()
list_countries = global_df["name"].unique()

In [67]:
list_indicators

array(['gini'], dtype=object)

In [83]:
for indicator in list_indicators:
    if(indicator == "Aid received"):
        continue
#for indicator in ["Completition rate female"]:
    #list_countries = global_df.loc[global_df["indicator"] == indicator,["name"]].unique()
    countries_with_no_data = []

    for country in list_countries:
    #for country in ["Iraq"]:
        min_year = global_df.loc[global_df["indicator"] == indicator, ["year"]].min().values[0]
        max_year =  global_df.loc[global_df["indicator"] == indicator, ["year"]].max().values[0]
        
        year_df_not_null = global_df.loc[~(global_df["value"].isnull()) 
                  & (global_df["indicator"] == indicator)
                  & (global_df["name"] == country), ["year"]].sort_values(by="year")
        
        list_year_not_null = [item for sublist in year_df_not_null.values for item in sublist]

        if(len(list_year_not_null) == 0):
            countries_with_no_data.append(country)
            continue
        
        # Mise à jour des premières valeurs si vides avec la première non vide
        #print(min_year)
        #print(max_year)
        #print(list_year_not_null)
        if(list_year_not_null[0] > min_year):
            #print("maj first values")
            value_first_indicator = float(global_df.loc[(global_df["indicator"] == indicator) 
                                      & (global_df["name"] == country) 
                                      & (global_df["year"] == list_year_not_null[0]), ["value"]].value)
            #print(value_first_indicator)
            for year in np.arange(min_year, list_year_not_null[0]):
                global_df.loc[(global_df["indicator"] == indicator) 
                                      & (global_df["name"] == country) 
                                      & (global_df["year"] == year), ["value"]] = value_first_indicator
            
   
        # Mise à jour des dernières valeurs si vides avec la dernière non vide
        if(list_year_not_null[-1] < max_year):
            #print("maj last values")
            value_last_indicator = float(global_df.loc[(global_df["indicator"] == indicator) 
                                      & (global_df["name"] == country) 
                                      & (global_df["year"] == list_year_not_null[-1]), ["value"]].value)
            #print(value_last_indicator)
            for year in np.arange(list_year_not_null[-1], max_year + 1):
                global_df.loc[(global_df["indicator"] == indicator) 
                                      & (global_df["name"] == country) 
                                      & (global_df["year"] == year), ["value"]] = value_last_indicator
    
    
        # Mise à jour des valeurs intermédiaires
        previous_year = list_year_not_null[0]
        nb_year_null = 0
        for current_year in list_year_not_null:
            if(current_year - previous_year > 1):
                nb_year_null = current_year - previous_year - 1
                value_previous_year = float(global_df.loc[(global_df["indicator"] == indicator) 
                                                     & (global_df["name"] == country) 
                                                     & (global_df["year"] == previous_year), ["value"]].value)
                value_current_year = float(global_df.loc[(global_df["indicator"] == indicator) 
                                                     & (global_df["name"] == country) 
                                                     & (global_df["year"] == current_year), ["value"]].value)
                step_value = (value_current_year - value_previous_year) / (nb_year_null + 1)

                for i, year in enumerate(np.arange(previous_year + 1, current_year)):
                    global_df.loc[(global_df["indicator"] == indicator) 
                                  & (global_df["name"] == country) 
                                  & (global_df["year"] == year), ["value"]] = value_previous_year + (i + 1) * step_value
            
            previous_year = current_year
            
    print(indicator)
    #if (len(countries_with_no_data) > 0):
    #   print("Pas de valeur pour l'indicteur : {}, pour les pays : {}".format(indicator, countries_with_no_data))

ValueError: could not convert string to float: ' China'

In [13]:
global_df_no_nan = global_df.copy()

In [14]:
global_df_no_nan.head()

Unnamed: 0  Unnamed: 0.1         name     indicator  \
0           0             0  Afghanistan  Aid received   
1           1             1  Afghanistan  Aid received   
2           2             2  Afghanistan  Aid received   
3           3             3  Afghanistan  Aid received   
4           4             4  Afghanistan  Aid received   

                                indicator2  year        value region  \
0  ODA received, total (constant 2010 US$)  1960  108800000.0   Asia   
1  ODA received, total (constant 2010 US$)  1961  223030000.0   Asia   
2  ODA received, total (constant 2010 US$)  1962  110970000.0   Asia   
3  ODA received, total (constant 2010 US$)  1963  225130000.0   Asia   
4  ODA received, total (constant 2010 US$)  1964  290570000.0   Asia   

  predicted  value_scaled  
0     False      3.445057  
1     False      7.169534  
2     False      3.780866  
3     False      3.744915  
4     False      3.728892

## Mise à l'échelle - min/max par année

In [33]:
file_path = "../static/data/scaled_indicators.csv"
global_df_scaled = pd.read_csv(file_path, sep=";")

In [40]:
def scale_value(value, min_val, max_val):
    scale_val = 100 * (value - min_val) / (max_val - min_val) + 25
    return scale_val

In [16]:
global_df_scaled = global_df_no_nan.copy()
global_df_scaled["value_scaled"] = global_df_scaled["value"]

In [35]:
del global_df_scaled["Unnamed: 0"]
del global_df_scaled["Unnamed: 0.1"]

In [38]:
indicator_to_rescale = global_df_scaled["indicator"].unique()

In [50]:
indicator_to_rescale

array(['Aid received', 'Exports (% of GDP)', 'External debt',
       'Total Gross Domestic Product (GDP) - Purchasing Power Parity (PPP)',
       'Aid received % of GNI (Gross National Income)',
       'Total number of billionaires',
       'Poverty headcount ratio at $2 a day (PPP) (% of population)',
       'Inflation', 'Investment', 'Tax revenue (% of GDP)',
       'Girls primary school completion rate (% of girls)',
       'Boys primary completion rate (% of boys)',
       'Primary completion rate - total (% of relevant age group)',
       'Expenditure per student - primary (% of GDP per capita)',
       'Expenditure per student - secondary (% of GDP per capita)',
       'Expenditure per student - tertiary (% of GDP per capita)',
       'Ratio of girls to boys in primary and secondary education (%)',
       'Years in school men 25+', 'Years in school women 25+',
       'Female 15-64 labour to population (%)',
       'Female 25-54 labour to population(%)',
       'Female 15+ labour 

In [85]:
(75-16)/8

7.375

In [89]:
np.linspace(16, 76, 10)

array([ 16.        ,  22.66666667,  29.33333333,  36.        ,
        42.66666667,  49.33333333,  56.        ,  62.66666667,
        69.33333333,  76.        ])

In [36]:
global_df_scaled.head()

name     indicator                                indicator2  year  \
0  Afghanistan  Aid received  ODA received - total (constant 2010 US$)  1960   
1  Afghanistan  Aid received  ODA received - total (constant 2010 US$)  1961   
2  Afghanistan  Aid received  ODA received - total (constant 2010 US$)  1962   
3  Afghanistan  Aid received  ODA received - total (constant 2010 US$)  1963   
4  Afghanistan  Aid received  ODA received - total (constant 2010 US$)  1964   

         value region predicted  value_scaled  
0  108800000.0   Asia     False      3.445057  
1  223030000.0   Asia     False      7.169534  
2  110970000.0   Asia     False      3.780866  
3  225130000.0   Asia     False      3.744915  
4  290570000.0   Asia     False      3.728892

In [41]:
#for indicator in list_indicators:
for indicator in indicator_to_rescale:
    df_one_indicator = global_df_scaled[global_df_scaled["indicator"] == indicator]
    
    min_year = global_df_scaled.loc[global_df_scaled["indicator"] == indicator, ["year"]].min().values[0]
    max_year =  global_df_scaled.loc[global_df_scaled["indicator"] == indicator, ["year"]].max().values[0]
    
    for year in np.arange(min_year, max_year + 1):
        min_value = global_df_scaled.loc[(global_df_scaled["year"] == year) 
                                         &(global_df_scaled["indicator"] == indicator),
                                         ["value"]].min().values[0]
        
        max_value = global_df_scaled.loc[(global_df_scaled["year"] == year) 
                                         &(global_df_scaled["indicator"] == indicator),
                                         ["value"]].max().values[0]
        
        global_df_scaled.loc[(global_df_scaled["indicator"] == indicator)
                             & (global_df_scaled["year"] == year),
                             ["value_scaled"]] = global_df_scaled[(global_df_scaled["indicator"] == indicator)
                             & (global_df_scaled["year"] == year)].apply(axis=1, func= lambda x: scale_value(x["value"], min_value, max_value))
        

In [49]:
global_df_scaled.loc[(global_df_scaled["indicator"] == "Boys primary completion rate (% of boys)")
                             & (global_df_scaled["year"] == 2000) & (global_df_scaled["value_scaled"] < 30)]

name                                 indicator  \
125467  Mozambique  Boys primary completion rate (% of boys)   
134952       Niger  Boys primary completion rate (% of boys)   
154930      Rwanda  Boys primary completion rate (% of boys)   

                                               indicator2  year     value  \
125467  Primary completion rate - male (% of relevant ...  2000  19.79628   
134952  Primary completion rate - male (% of relevant ...  2000  22.30817   
154930  Primary completion rate - male (% of relevant ...  2000  24.72564   

        region predicted  value_scaled  
125467  Africa     False     25.000000  
134952  Africa     False     27.064611  
154930  Africa     False     29.051615

In [42]:
global_df_scaled.loc[(global_df_scaled["indicator"] == "Boys primary completion rate (% of boys)")
                             & (global_df_scaled["year"] == 2000),["value_scaled"]].describe()

/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/numpy/lib/function_base.py:4116: RuntimeWarning: Invalid value encountered in percentile
  interpolation=interpolation)


value_scaled
count    166.000000
mean      76.177920
std       18.895117
min       25.000000
25%             NaN
50%             NaN
75%             NaN
max      125.000000

## Enregistrement du csv

In [28]:
global_df_scaled[(global_df["value"].isnull())]

name                                          indicator  \
423          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
424          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
425          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
426          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
427          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
428          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
429          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
430          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
431          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
432          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
433          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
434          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
435          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
436          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
437          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
438          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
439          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
440          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
441          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
442          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
443          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
444          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
445          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
446          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
447          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
448          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
449          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
450          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
451          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
452          Afghanistan  Poverty headcount ratio at $2 a day (PPP) (% o...   
...                  ...                                                ...   
204486  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204487  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204488  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204489  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204490  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204491  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204492  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204493  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204494  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204495  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204496  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204497  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204498  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204499  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204500  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204501  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204502  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204503  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasing...   
204504  Saint Barth̩lemy  Total Gross Domestic Product (GDP), Purchasi

In [43]:
global_df_scaled.to_csv("scaled_indicators_bis.csv", sep=";")

## Analyse des dates pour savoir lesquelles conserver

In [39]:
for indicator in list_indicators:
    min_year = global_df.loc[global_df["indicator"] == indicator, ["year"]].min().values[0]
    max_year =  global_df.loc[global_df["indicator"] == indicator, ["year"]].max().values[0]
    print("Traitement de l'indicateur : {}, de l'année {} à {}".format(indicator, min_year, max_year))
    #global_df.loc[~(global_df["value"].isnull()) & (global_df["indicator"] == indicator), ["year"]].sort_values()

Traitement de l'indicateur : Aid received, de l'année 1960 à 2010
Traitement de l'indicateur : Exports (p of GDP), de l'année 1961 à 2011
Traitement de l'indicateur : External debt, de l'année 1960 à 2011
Traitement de l'indicateur : indicator gdp_total_ppp, de l'année 1800 à 2013
Traitement de l'indicateur : indicator_aid received % of gni, de l'année 1960 à 2010
Traitement de l'indicateur : Indicator_number of billionaires, de l'année 2004 à 2007
Traitement de l'indicateur : Indicator_Poverty headcount ratio at $2 a day (PPP) (% of population), de l'année 1978 à 2010
Traitement de l'indicateur : Inflation, de l'année 1961 à 2011
Traitement de l'indicateur : Investment, de l'année 1960 à 2011
Traitement de l'indicateur : Tax revenue (p of GDP), de l'année 1990 à 2011
Traitement de l'indicateur : Completition rate female, de l'année 1970 à 2011
Traitement de l'indicateur : Completition rate male, de l'année 1970 à 2011
Traitement de l'indicateur : Completition rate total, de l'année 19

In [88]:
global_df.loc[(global_df["indicator"] == "Completition rate female")
              & (global_df["name"] == "Iraq"), ["year", "value"]].sort_values(by="year")

year     value
86661  1970       NaN
86662  1971  20.56606
86663  1972  20.85348
86664  1973  22.79115
86665  1974  23.31958
86666  1975  27.31529
86667  1976  28.56560
86668  1977  37.18561
86669  1978  41.62401
86670  1979       NaN
86671  1980  53.15110
86672  1981       NaN
86673  1982       NaN
86674  1983  48.47658
86675  1984       NaN
86676  1985  58.33163
86677  1986  56.85311
86678  1987  53.14240
86679  1988  57.77889
86680  1989  53.36140
86681  1990       NaN
86682  1991       NaN
86683  1992       NaN
86684  1993  54.00860
86685  1994       NaN
86686  1995       NaN
86687  1996       NaN
86688  1997       NaN
86689  1998       NaN
86690  1999  50.87453
86691  2000  50.08348
86692  2001  54.36886
86693  2002       NaN
86694  2003  68.21609
86695  2004  67.17209
86696  2005       NaN
86697  2006       NaN
86698  2007  55.39513
86699  2008       NaN
86700  2009       NaN
86701  2010       NaN
86702  2011       NaN

In [117]:
global_df.loc[(global_df["indicator"] == "Completition rate female")
              & (global_df["name"] == "Iraq"), ["year", "value"]].sort_values(by="year")

year      value
86661  1970  20.566060
86662  1971  20.566060
86663  1972  20.853480
86664  1973  22.791150
86665  1974  23.319580
86666  1975  27.315290
86667  1976  28.565600
86668  1977  37.185610
86669  1978  41.624010
86670  1979  47.387555
86671  1980  53.151100
86672  1981  51.592927
86673  1982  50.034753
86674  1983  48.476580
86675  1984  53.404105
86676  1985  58.331630
86677  1986  56.853110
86678  1987  53.142400
86679  1988  57.778890
86680  1989  53.361400
86681  1990  53.523200
86682  1991  53.685000
86683  1992  53.846800
86684  1993  54.008600
86685  1994  53.486255
86686  1995  52.963910
86687  1996  52.441565
86688  1997  51.919220
86689  1998  51.396875
86690  1999  50.874530
86691  2000  50.083480
86692  2001  54.368860
86693  2002  61.292475
86694  2003  68.216090
86695  2004  67.172090
86696  2005  63.246437
86697  2006  59.320783
86698  2007  55.395130
86699  2008  55.395130
86700  2009  55.395130
86701  2010  55.395130
86702  2011  55.395130

In [55]:
year_df = global_df.loc[~(global_df["value"].isnull()) 
              & (global_df["indicator"] == "Aid received")
              & (global_df["name"] == "Germany"), ["year"]].sort_values(by="year")
list_year = [item for sublist in year_df.values for item in sublist]

In [57]:
global_df[(global_df["indicator"] == "Aid received")
              & (global_df["name"] == "Germany")]

Unnamed: 0     name     indicator  \
66262       66262  Germany  Aid received   
66263       66263  Germany  Aid received   
66264       66264  Germany  Aid received   
66265       66265  Germany  Aid received   
66266       66266  Germany  Aid received   
66267       66267  Germany  Aid received   
66268       66268  Germany  Aid received   
66269       66269  Germany  Aid received   
66270       66270  Germany  Aid received   
66271       66271  Germany  Aid received   
66272       66272  Germany  Aid received   
66273       66273  Germany  Aid received   
66274       66274  Germany  Aid received   
66275       66275  Germany  Aid received   
66276       66276  Germany  Aid received   
66277       66277  Germany  Aid received   
66278       66278  Germany  Aid received   
66279       66279  Germany  Aid received   
66280       66280  Germany  Aid received   
66281       66281  Germany  Aid received   
66282       66282  Germany  Aid received   
66283       66283  Germany  Aid received   
66284       66284  Germany  Aid received   
66285       66285  Germany  Aid received   
66286       66286  Germany  Aid received   
66287       66287  Germany  Aid received   
66288       66288  Germany  Aid received   
66289       66289  Germany  Aid received   
66290       66290  Germany  Aid received   
66291       66291  Germany  Aid received   
66292       66292  Germany  Aid received   
66293       66293  Germany  Aid received   
66294       66294  Germany  Aid received   
66295       66295  Germany  Aid received   
66296       66296  Germany  Aid received   
66297       66297  Germany  Aid received   
66298       66298  Germany  Aid received   
66299       66299  Germany  Aid received   
66300       66300  Germany  Aid received   
66301       66301  Germany  Aid received   
66302       66302  Germany  Aid received   
66303       66303  Germany  Aid received   
66304       66304  Germany  Aid received   
66305       66305  Germany  Aid received   
66306       66306  Germany  Aid received   
66307       66307  Germany  Aid received   
66308       66308  Germany  Aid received   
66309       66309  Germany  Aid received   
66310       66310  Germany  Aid received   
66311       66311  Germany  Aid received   
66312       66312  Germany  Aid received   

                                    indicator2  year  value  region predicted  
66262  ODA received, total (constant 2010 US$)  1960    NaN  Europe      True  
66263  ODA received, total (constant 2010 US$)  1961    NaN  Europe      True  
66264  ODA received, total (constant 2010 US$)  1962    NaN  Europe      True  
66265  ODA received, total (constant 2010 US$)  1963    NaN  Europe      True  
66266  ODA received, total (constant 2010 US$)  1964    NaN  Europe      True  
66267  ODA received, total (constant 2010 US$)  1965    NaN  Europe      True  
66268  ODA received, total (constant 2010 US$)  1966    NaN  Europe      True  
66269  ODA received, total (constant 2010 US$)  1967    NaN  Europe      True  
66270  ODA received, total (constant 2010 US$)  1968    NaN  Europe      True  
66271  ODA received, total (constant 2010 US$)  1969    NaN  Europe      True  
66272  ODA received, total (constant 2010 US$)  1970    NaN  Europe      True  
66273  ODA received, total (constant 2010 US$)  1971    NaN  Europe      True  
66274  ODA received, total (constant 2010 US$)  1972    NaN  Europe      True  
66275  ODA received, total (constant 2010 US$)  1973    NaN  Europe      True  
66276  ODA received, total (constant 2010 US$)  1974    NaN  Europe      True  
66277  ODA received, total (constant 2010 US$)  1975    NaN  Europe      True  
66278  ODA received, total (constant 2010 US$)  1976    NaN  Europe      True  
66279  ODA received, total (constant 2010 US$)  1977    NaN  Europe      True  
66280  ODA received, total (constant 2010 US$)  1978    NaN  Europe      True  
66281  ODA received, total (constant 2010 US$)  1979    NaN  Europe      True  
66282  ODA received, total (constant 2

In [41]:
global_df[(global_df["indicator"] == "expenditure primary")
              & (global_df["name"] == "France")]

Unnamed: 0    name            indicator  \
60727       60727  France  expenditure primary   
60728       60728  France  expenditure primary   
60729       60729  France  expenditure primary   
60730       60730  France  expenditure primary   
60731       60731  France  expenditure primary   
60732       60732  France  expenditure primary   
60733       60733  France  expenditure primary   
60734       60734  France  expenditure primary   
60735       60735  France  expenditure primary   
60736       60736  France  expenditure primary   
60737       60737  France  expenditure primary   
60738       60738  France  expenditure primary   
60739       60739  France  expenditure primary   
60740       60740  France  expenditure primary   

                                              indicator2  year     value  \
60727  Expenditure per student, primary (% of GDP per...  1998       NaN   
60728  Expenditure per student, primary (% of GDP per...  1999  17.88172   
60729  Expenditure per student, primary (% of GDP per...  2000  17.95715   
60730  Expenditure per student, primary (% of GDP per...  2001  17.96406   
60731  Expenditure per student, primary (% of GDP per...  2002  18.23649   
60732  Expenditure per student, primary (% of GDP per...  2003  18.17383   
60733  Expenditure per student, primary (% of GDP per...  2004  18.42795   
60734  Expenditure per student, primary (% of GDP per...  2005  17.98432   
60735  Expenditure per student, primary (% of GDP per...  2006  17.64613   
60736  Expenditure per student, primary (% of GDP per...  2007  18.29524   
60737  Expenditure per student, primary (% of GDP per...  2008  18.01231   
60738  Expenditure per student, primary (% of GDP per...  2009       NaN   
60739  Expenditure per student, primary (% of GDP per...  2010       NaN   
60740  Expenditure per student, primary (% of GDP per...  2011       NaN   

       region predicted  
60727  Europe      True  
60728  Europe     False  
60729  Europe     False  
60730  Europe     False  
60731  Europe     False  
60732  Europe     False  
60733  Europe     False  
60734  Europe     False  
60735  Europe     False  
60736  Europe     False  
60737  Europe     False  
60738  Europe      True  
60739  Europe      True  
60740  Europe      True